### PR TITLE
BR-13 Implement @PostConstruct

### PR DESCRIPTION
**Pull request by Pan14ek**
### Issue/Task

[BR-13 Implement @PostConstruct, @PreDestroy](https://trello.com/c/rO9P8YmV)

### Description

Added the `PostConstruct` annotation to invoke methods after initializing beans. It can throw an exception `NoUniquePostConstructException` if the class has more than one method annotated with `PostConstruct`.

### Testing

Wrote unit tests to test using the `PostConstruct` annotation.

### CODE REVIEW

**Review by JJJazl — APPROVED** [review]

**Review by rovein — APPROVED** [review]
LGTM!

File: src/main/java/svydovets/core/context/DefaultApplicationContext.java [hunk 1]
```diff
@@ -97,8 +99,44 @@ private Object initWithBeanPostProcessor(Map.Entry<String, BeanDefinition> entry
...
+        Arrays.stream(declaredMethods)
+                .filter(isAnnotatedMethod)
+                .findFirst()
+                .ifPresent(method -> {
+                    try {
+                        method.setAccessible(true);
+                        method.invoke(bean);
+                    } catch (IllegalAccessException | InvocationTargetException e) {
+                        throw new InvalidInvokePostConstructMethodException("Something went wrong. Please check the method that was annotated with @PostConstruct", e);
+                    }
```
Comment (JJJazl) [inline src/main/java/svydovets/core/context/DefaultApplicationContext.java:138]: Move it to additional method, please. Block code in lambdas is not best practice

File: src/main/java/svydovets/core/context/DefaultApplicationContext.java [hunk 2]
```diff
@@ -97,8 +99,44 @@ private Object initWithBeanPostProcessor(Map.Entry<String, BeanDefinition> entry
...
+     */
     private void postConstructInitialization(Object bean) {
-        throw new UnsupportedOperationException();
+        Class<?> beanType = bean.getClass();
+        Method[] declaredMethods = beanType.getDeclaredMethods();
+        Predicate<Method> isAnnotatedMethod = method -> method.isAnnotationPresent(PostConstruct.class);
+
+        var filteredMethods = Arrays.stream(declaredMethods).filter(isAnnotatedMethod);
+
+        boolean isNotUniqueMethod = filteredMethods.count() > 1;
```
Comment (JJJazl) [inline src/main/java/svydovets/core/context/DefaultApplicationContext.java:123]: You can simplify it:
boolean isNotUniqueMethod = Arrays.stream(declaredMethods)
.filter(method -> method.isAnnotationPresent(PostConstruct.class))
.count() > 1

Comment (Pan14ek) [reply src/main/java/svydovets/core/context/DefaultApplicationContext.java:123]: Fixed in the PR https://github.com/Svydovets-Bobocode-Java-Ultimate-3-0/Bring/pull/10

File: src/main/java/svydovets/core/context/DefaultApplicationContext.java (hunk 1 above)
Comment (Pan14ek) [reply src/main/java/svydovets/core/context/DefaultApplicationContext.java:138]: Fixed in the PR https://github.com/Svydovets-Bobocode-Java-Ultimate-3-0/Bring/pull/10

**Pull request merged**